pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentUploadRestletResource.java

```diff
@@ -288,7 +288,15 @@ private SuccessResponse uploadSegment(FormDataMultiPart multiPart, boolean enabl
           throw new UnsupportedOperationException("Unsupported upload type: " + uploadType);
       }
 
-      String rawTableName = segmentMetadata.getTableName();
+      // Fetch raw table name. Try to derive the table name from the header and then from segment metadata
```

Comment:
We plan to include upload of merged realtime segments from minion, right? So, we should include typed table name in the headers, not raw table name

Reply:
Current implementation reads `rawTableName` and we have written `rawTableName` to the segment metadata. I think that we should keep it in the same behavior for now.

We can design it correctly when we implement the upload path for the realtime segments.

Reply:
Current segment upload API provides the table header only as an information. It is (or should not be) used anywhere other than logs. Do you see it used any place?

Reply:
Agreed that we can add a new table type header for realtime. thanks.